### PR TITLE
exclude generated swagger files from coverage

### LINKED ISSUE #15: exclude auto generated files from test coverage
certain files might be generated by tools like openapi-cli, e.g.:
- swagger.validator.ts
- swagger.guard.ts
they should be excluded from the test coverage otherwise it will impact the coverage report negatively

### PR DESCRIPTION
**Pull request by le-cong**
Fixes #15

### CODE REVIEW

**Comment by carlansley** [comment]
those files are prod code (even though generated).. might a reduction in coverage potentially be an issue?

**Comment by carlansley** [comment]
also, the `publish-beta.yml` should look like:

```
name: Publish Beta Package

on:
  pull_request:
    branches:
      - 'main'

jobs:
  publishBeta:
    runs-on: ubuntu-latest
    name: Publish Beta
    strategy:
      matrix:
        node-version: [18.x]
    steps:
      - uses: actions/checkout@v3
        with:
          ref: ${{ github.event.pull_request.head.sha }}
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v3
        with:
          node-version: ${{ matrix.node-version }}
          registry-url: 'https://registry.npmjs.org'
      - name: Install dependencies
        run: npm ci --ignore-scripts
      - name: Audit Signatures
        run: npm audit signatures
      - name: Run Test
        run: npm run test
      - name: Prepare Beta
        uses: checkdigit/github-actions/prepare-beta@main
      - name: Publish Beta
        run: npm publish --tag beta
        env:
          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN_PUBLISH }}
```

**Comment by carlansley** [comment]
the `publish-beta` branch has been deleted/squerged into `main`

**Comment by le-cong** [comment]
> those files are prod code (even though generated).. might a reduction in coverage potentially be an issue?

good point, really need to think about it. e.g. things like static validator / type guard are indeed prod code, but on the other hand the test coverage could drop dramatically. 
can those ajv generated code consider stable/safe 🤔

**Comment by le-cong** [comment]
had to sync the "publish-beta.yml" config file from sentinel, now it published beta correctly

File: .github/workflows/publish-beta.yml [hunk 1]
```diff
@@ -20,16 +20,20 @@ jobs:
...
         run: npm audit signatures
       - name: Run Test
-        run: npm run ci:test
-      - name: Prepare Beta
-        uses: checkdigit/github-actions/prepare-beta@prepare-beta
-      - name: Publish Beta
-        run: npm publish --tag beta
+        run: npm run test
+      - name: perform beta publish
+        uses: checkdigit/github-actions/publish-beta@main
```
Comment (carlansley) [inline .github/workflows/publish-beta.yml]: I think this file should be reverted.  if you just use this `uses` line from the original file it should be fine?

File: .npmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+//registry.npmjs.org/:_authToken=${NPM_TOKEN}
```
Comment (carlansley) [inline .npmrc]: this file isn't needed if the original publish-beta.yml is used

File: jest-preset.js [hunk 3]
```diff
@@ -21,7 +21,14 @@ export default {
...
     ],
   },
-  collectCoverageFrom: ['<rootDir>/src/**', '!<rootDir>/src/**/*.spec.ts', '!<rootDir>/src/**/*.test.ts'],
+  collectCoverageFrom: [
+    '<rootDir>/src/**',
+    '!<rootDir>/src/**/*.spec.ts',
+    '!<rootDir>/src/**/*.test.ts',
+    '!<rootDir>/src/**/swagger.ts',
+    '!<rootDir>/src/**/swagger.validator.ts',
+    '!<rootDir>/src/**/swagger.guard.ts',
```
Comment (carlansley) [inline jest-preset.js:30]: I'm still uncomfortable with this change.  it's oddly specific to openapi for a general-purpose jest-config imo.  need to marinate on it more.

**Review by carlansley — CHANGES_REQUESTED** [review]

File: jest-preset.js (hunk 3 above)
Comment (le-cong) [reply jest-preset.js:30]: that makes sense

Comment (le-cong) [reply jest-preset.js:30]: even `swagger.ts` is managed in each repo's `.gitignore`
totally agree that it's not ideal to push special stuff into general purpose library.
cloing this PR now.

btw I was confused by the publish-beta.yml, just realized that it's using a bit different approach comparing to existing services :)

Comment (carlansley) [reply jest-preset.js:30]: yeah this uses the new tag/release publish workflow :)

**Pull request closed without merging**